ci-cd/github-actions.md

```diff
@@ -0,0 +1,218 @@
+2. Set up the GitHub Actions workflow
+3. Optimize your workflow to reduce the number of pull requests and the total build time, and finally,
+4. Push only specific versions to Docker Hub.
+
+## Set up a Docker project
+
+Let’s get started. This guide uses a simple Docker project as an example. The [SimpleWhaleDemo](https://github.com/usha-mandya/SimpleWhaleDemo){:target="_blank" rel="noopener" class="_"} repository contains an Ngnix alpine image. You can either clone this repository, or use your own Docker project.
+
+![SimpleWhaleDemo](images/simplewhaledemo.png){:width="500px"}
+
+Before we start, ensure you can access [Docker Hub](https://hub.docker.com/) from any workflows you create. To do this:
+
+1. Add your Docker ID and Personal Access Token (PAT) as secrets to GitHub.
+2. To create a new token, go to [Docker Hub Settings](https://hub.docker.com/settings/security) and then click **New Access Token**.
+3. Let’s call this token **simplewhaleci**.
+
+    ![New access token](images/github-access-token.png){:width="500px"}
+
+4. Now, add this token to your username as secrets into the GitHub secrets UI.
+
+    ![GitHub Secrets](images/github-secrets.png){:width="500px"}
+
+## Set up the GitHub Actions workflow
+
+In the previous section, we created a PAT and added it to GitHub to ensure we can access Docker Hub from any workflow. Now, let’s set up our GitHub Actions workflow to build and store our images in Hub. We can achieve this by creating two Docker actions:
+
+1. The first action enables us to log in to Docker Hub using the secrets we stored in the GitHub Repository.
+2. The second one is the build and push action.
+
+In this example, let us set the push flag to `true` as we also want to push. We’ll then add a tag to specify to always go to the latest version. Lastly, we’ll echo the image digest to see what was pushed.
+
+To set up the workflow:
+
+1. Go to your repository in GitHub and then click **Actions** > **New workflow**.
+2. Click **set up a workflow yourself** and add the following content:
+
+First, we will name this workflow:
+
+```yaml
+name: CI to Docker Hub
+ ```
+
+Then, we will choose when we run this workflow. In our example, we are going to do it for every push against the main branch of our project:
+
+```yaml
+on:
+  push:
+    branches: [ main ]
+```
+
+Now, we need to specify what we actually want to happen within our action (what jobs), we are going to add our build one and select that it runs on the latest Ubuntu instances available:
+
+```yaml
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+```
+
+Now, we can add the steps required. The first one is to use our PAT and username to log into Docker Hub. The second is the Builder, the action  uses BuildKit under the hood through a simple Buildx action which we will also setup
+
+```yaml
+    Steps:
+
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
```

Comment:
Oh, these double curly braces here and further down in the code samples get lost in Jekyll. 😢 

![docs-yaml-vars-missing](https://user-images.githubusercontent.com/207759/97193300-2570e680-17a9-11eb-9523-e1c54ca6faa9.jpg)

Looks like there are some workarounds to escape the curly braces https://stackoverflow.com/questions/24102498/escaping-double-curly-braces-inside-a-markdown-code-block-in-jekyll

Reply:
I've added `{% raw %}` and `{% endraw %}` tags

Reply:
![image](https://user-images.githubusercontent.com/47779042/97238697-ead77000-17e1-11eb-85eb-6d75f6b2f18b.png)